src/figuresClasses.ts

```diff
@@ -1,11 +1,79 @@
-export interface Figure {}
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: Figure['shape'] = 'triangle';
+
+  constructor(
+    public color: Figure['color'],
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const sides: number[] = [a, b, c];
+
+    if (sides.some((s) => s <= 0)) {
+      throw new Error('Invalid sides');
+    }
 
-export class Triangle implements Figure {}
+    const largest: number = Math.max(...sides);
 
-export class Circle implements Figure {}
+    sides.splice(sides.indexOf(Math.max(...sides)), 1);
+
+    if (largest >= sides.reduce((acc, cur) => acc + cur)) {
+      throw new Error(`sides ${a}, ${b} and ${c} can't form a triangle`);
+    }
+  }
+
+  getArea(): number {
```

Comment:
[CODE STYLE] - The `getArea` function should not accept any parameters as per the checklist. All necessary values are already available as object fields. Please remove any parameters and use the class fields directly.